[FreeBSD] macro compilation fail (evolution test)

### Check duplicate issues.

- [ ] Checked for duplicates

### Description

Something goes wrong in the code generation for a lot of tests, this case is roottest-root-meta-evolution-data1-build
for some reason the first entry from the list of available elf objects ([vdso]) makes it to the generated command line (and because it is not a path, I imagine, something get's thrown off). Where is that generated? A quick search left me with nothing... I suspect this is a reason for a lot of trouble here.

```
1874: Processing /home/mkrzewic/src/root/roottest/scripts/build.C("/home/mkrzewic/src/root/roottest/root/meta/evolution/data1.C","","")...
1874: Info in <TUnixSystem::ACLiC>: creating shared library /home/mkrzewic/build/root/roottest/root/meta/evolution/data1_C.so
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82394b8f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x82ce907b0 from unloaded dso, skipping
1874: sh: Syntax error: Unterminated quoted string
1874: Error in <ACLiC>: Executing 'cd "/home/mkrzewic/build/root/roottest/root/meta/evolution" ; c++ -fPIC -c -g  -std=c++17 -Wc++11-narrowing -Wsign-compare -Wsometimes-uninitialized -Wconditional-uninitialized -Wheader-guard -Warray-bounds -Wcomment -Wtautological-compare -Wstrncat-size -Wloop-analysis -Wbool-conversion -D_REENTRANT -pipe -W -Woverloaded-virtual -fsigned-char -pthread -DClingWorkAroundTClassUpdateDouble32   -I/usr/local/include/root -I/home/mkrzewic/build/root/roottest/root/meta/evolution -I"/usr/home/mkrzewic/build/root/etc/" -I"/usr/home/mkrzewic/build/root/etc//cling" -I"/usr/home/mkrzewic/build/root/etc//cling/plugins/include" -I"/usr/home/mkrzewic/build/root/include/" -I"/usr/home/mkrzewic/build/root/include" -I"/home/mkrzewic/build/root/roottest/root/meta/evolution"   -D__ACLIC__ "/home/mkrzewic/build/root/roottest/root/meta/evolution/data1_C_ACLiC_dict.cxx" ; c++ -g  "/home/mkrzewic/build/root/roottest/root/meta/evolution/data1_C_ACLiC_dict.o" -shared  -rdynamic  "-lRint" "-lCore" "-L/usr/local/lib/root" "[vdso]  -o "/home/mkrzewic/build/root/roottest/root/meta/evolution/data1_C.so"' failed!
1874: __cxa_thread_call_dtors: dtr 0x824a907b0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x8231078f0 from unloaded dso, skipping
1874: __cxa_thread_call_dtors: dtr 0x8231078f0 from unloaded dso, skipping
1874: gmake[1]: *** [roottest/root/meta/evolution/CMakeFiles/roottest-root-meta-evolution-data1-compile-macro.dir/build.make:70: roottest/root/meta/evolution/CMakeFiles/roottest-root-meta-evolution-data1-compile-macro] Error 1
1874: gmake: *** [Makefile:25710: roottest-root-meta-evolution-data1-compile-macro/fast] Error 2
1/1 Test #1874: roottest-root-meta-evolution-data1-build ...***Failed    5.71 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   6.86 sec

The following tests FAILED:
1874 - roottest-root-meta-evolution-data1-build (Failed)
Errors while running CTest
Output from these tests are in: /home/mkrzewic/build/root/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

### Reproducer

ctest -j4 -R "roottest-root-meta-evolution-data1-build" -V

### ROOT version

master

### Installation method

source

### Operating system

FreeBSD 13.2

### Additional context

_No response_

--- a/core/metacling/src/TCling.cxx
+++ b/core/metacling/src/TCling.cxx
@@ -3280,7 +3280,14 @@ static int callback_for_dl_iterate_phdr(struct dl_phdr_info *info, size_t size, 
       // Skip \0, "", and kernel pseudo-libs linux-vdso.so.1 or linux-gate.so.1
       if (info->dlpi_name && info->dlpi_name[0]
 #if defined(R__FBSD)
-          && strstr(info->dlpi_name, "[vdso]")
+          //skip the executable (with null addr)
+          && info->dlpi_addr
+          //has no path
+          && strncmp(info->dlpi_name, "[vdso]", 6)
+          //the linker does not like to be mmapped
+          //causes a crash in cling::DynamicLibraryManager::loadLibrary())
+          //with error message "mmap of entire address space failed: Cannot allocate memory"
+          && strncmp(info->dlpi_name, "/libexec/ld-elf.so.1", 20)
 #endif
           && strncmp(info->dlpi_name, "linux-vdso.so", 13)
           && strncmp(info->dlpi_name, "linux-vdso32.so", 15)

